packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/state_vars/immediate_value.rs

```diff
@@ -4,93 +4,90 @@ use super::TextInputState;
-        self.dependency_instructions = ImmediateValueDependencyInstructions {
-            essential: Some(DependencyInstruction::Essential),
-            sync_immediate_value: Some(
-                TextInputState::get_sync_immediate_value_dependency_instructions(),
-            ),
-            bind_value_to: Some(TextInputState::get_bind_value_to_dependency_instructions()),
-            prefill: Some(TextInputState::get_prefill_dependency_instructions()),
+    ) -> Vec<DataQuery> {
+        self.data_queries = RequiredDataDataQueries {
+            essential: Some(DataQuery::Essential),
+            sync_immediate_value: Some(TextInputState::get_sync_immediate_value_data_queries()),
+            bind_value_to: Some(TextInputState::get_bind_value_to_data_queries()),
+            prefill: Some(TextInputState::get_prefill_data_queries()),
         };
 
-        (&self.dependency_instructions).into()
+        (&self.data_queries).into()
     }
 
-    fn save_dependencies(&mut self, dependencies: &Vec<DependenciesCreatedForInstruction>) {
-        self.dependency_values = dependencies.try_into().unwrap();
+    fn save_query_results(&mut self, dependencies: &Vec<DependenciesCreatedForInstruction>) {
+        self.query_results = dependencies.try_into().unwrap();
     }
 
-    fn calculate_state_var_from_dependencies(&self) -> StateVarCalcResult<String> {
-        let immediate_value = if !self.dependency_values.bind_value_to.came_from_default()
-            && *self.dependency_values.sync_immediate_value.get()
+    fn calculate(&self) -> StateVarCalcResult<String> {
+        let immediate_value = if !self.query_results.bind_value_to.came_from_default()
+            && *self.query_results.sync_immediate_value.get()
         {
-            self.dependency_values.bind_value_to.get().clone()
-        } else if self.dependency_values.essential.came_from_default() {
-            self.dependency_values.prefill.get().clone()
+            self.query_results.bind_value_to.get().clone()
+        } else if self.query_results.essential.came_from_default() {
+            self.query_results.prefill.get().clone()
         } else {
-            self.dependency_values.essential.get().clone()
+            self.query_results.essential.get().clone()
         };
         StateVarCalcResult::Calculated(immediate_value)
     }
 
-    fn request_dependency_updates(
+    fn invert(
         &mut self,
-        state_var: &StateVarReadOnlyView<String>,
+        state_var: &StateVarView<String>,
         is_direct_change_from_renderer: bool,
     ) -> Result<Vec<DependencyValueUpdateRequest>, RequestDependencyUpdateError> {
         let requested_value = state_var.get_requested_value();
 
-        let bind_value_to_came_from_default =
-            self.dependency_values.bind_value_to.came_from_default();
+        let bind_value_to_came_from_default = self.query_results.bind_value_to.came_from_default();
 
-        self.dependency_values
+        self.query_results
```

Comment:
The term `query_results` seem funny here.

Maybe it's confusing as it seems like a verb, that you are trying to "query the results".

What if "required data" becomes something like "input data" (not the best term) so that it becomes `self.input_data.name.queue_update()`.

I found the term "required" confusing at first, as I was thinking it was going to be followed by "optional" data.

I can't think of another term besides "input" or "dependent".

Reply:
It could just be `data` or `data_cache`.

Reply:
If it is `data_cache`, then does `queue_update()` on the cache make sense? Maybe just `data` then. Not entirely sure if queuing an update on the `data` makes sense, but I can't think of a better term.

Reply:
But we're not doing `data.queue_update`, we're doing `data.<statevar>.queue_update` Queuing on a state var makes sense, I think.

Reply:
OK. I think data is better than `query_results`.